mantis/simulation/routers/lautaro.py

```diff
@@ -0,0 +1,201 @@
+# solves using NLP optimization (or what best underlying engine decides)
+# Models cross chain transfers as fees as """pools"""
+# Uses decision variables to decide if to do Transfer to tap pool or not.
+
+import numpy as np
+import cvxpy as cp
+import copy 
+import threading as th
```

Comment:
Th does no work in cpyhon. Is pypy needed?

Reply:
I haven't tried to use other enviorments. I only used poetry run

Reply:
threading in cpython as of now always single threaded 

```
CPython implementation detail: In CPython, due to the [Global Interpreter Lock](https://docs.python.org/3/glossary.html#term-global-interpreter-lock), only one thread can execute Python code at once (even though certain performance-oriented libraries might overcome this limitation). If you want your application to make better use of the computational resources of multi-core machines, you are advised to use [multiprocessing](https://docs.python.org/3/library/multiprocessing.html#module-multiprocessing) or [concurrent.futures.ProcessPoolExecutor](https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.ProcessPoolExecutor). However, threading is still an appropriate model if you want to run multiple I/O-bound tasks simultaneously.
```